joblib/test/test_parallel.py

```diff
@@ -85,6 +91,40 @@ def f(x, y=0, z=0):
     return x ** 2 + y + z
 
 
+def check_subprocess_call(cmd, timeout=1, stdout_regex=None):
+    """Runs a command in a subprocess with timeout in seconds.
+
+    Also checks returncode is zero and stdout if stdout_regex is set.
+    """
+    proc = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE)
+
+    def kill_process():
+        proc.kill()
+
+    timer = Timer(timeout, kill_process)
+    try:
+        timer.start()
+        stdout, stderr = proc.communicate()
+
+        if PY3:
+            stdout, stderr = stdout.decode(), stderr.decode()
+        if proc.returncode != 0:
+            message = (
+                'Non-zero return code: {0}.\nStdout is:\n{0}\n'
+                'Stderr is:\n{2}').format(
+                    proc.returncode, stdout, stderr)
+            raise ValueError(message)
```

Comment:
Yes, that would be great because it can be helpful.

I tried the following at the end of the test function and could verify `check_subprocess_call` works as expected:
```python
nose.tools.assert_raises(ValueError,
                         check_subprocess_call,
                         [sys.executable, '-c', code],
                         stdout_regex=r'\[0, 1, 9, 16\]')

wrong_code = '\n\n'.join([
    'from joblib import Parallel, delayed',
    'def sqrt(x): return y**2',
    'print(Parallel(n_jobs=2)(delayed(sqrt)(i) for i in range(5)))'])

nose.tools.assert_raises(ValueError,
                         check_subprocess_call,
                         [sys.executable, '-c', wrong_code],
                         stdout_regex=r'\[0, 1, 4, 9, 16\]')
```